src/format/mod.rs

```diff
@@ -386,10 +437,12 @@ pub fn format_item<'a>(
     time: Option<&NaiveTime>,
     off: Option<&(String, FixedOffset)>,
     item: &Item<'a>,
-) -> fmt::Result {
+    locale: &str,
+) -> FormatResult<()> {
```

Comment:
we should create a new `fn format_item_localized` function to preserve backwards compatibility. I also think that for backcompat the default locale should be `"en_US"` or `"en_GB"`.

That also allows us to return the improved `FormatResult` from that function, but not change the error type here (which is also backwards-incompatible)

Reply:
I think we should actually use POSIX for the default https://docs.rs/pure-rust-locales/0.2.1/pure_rust_locales/POSIX/LC_TIME/index.html ... but maybe it doesn't change anything with en_US https://docs.rs/pure-rust-locales/0.2.1/pure_rust_locales/en_US/LC_TIME/index.html

Reply:
Oh yes POSIX makes more sense! Aggreed.

Reply:
Done! More or less... I did duplicate some function but I broke the compatibility since I changed the result `fmt::Result` to `FormatResult`